datafusion/sql/src/statement.rs

```diff
@@ -514,6 +517,35 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                     return not_impl_err!("To not supported")?;
                 }
 
+                // put the statement back together temporarily to get the SQL
```

Comment:
While this formulation is more code, it does avoid serializing the entire input string on every query, even when it is not needed